src/raw_inputs.rs

```diff
@@ -0,0 +1,163 @@
+//! Raw input events that make up a user input.
+
+use std::hash::Hash;
+
+use bevy::prelude::{GamepadAxisType, GamepadButtonType, KeyCode, MouseButton};
+use itertools::Itertools;
+
+use crate::axislike::DualAxisType;
+use crate::user_input::{GamepadControlDirection, MouseMoveDirection, MouseScrollDirection};
+
+/// The basic input events that make up a [`UserInput`](crate::user_input::UserInput).
+///
+/// Typically obtained by calling [`UserInput::raw_inputs`](crate::user_input::UserInput::raw_inputs).
+#[derive(Default, Debug, Clone, PartialEq)]
+#[must_use]
+pub struct RawInputs {
```

Comment:
Hmm. I don't love this design. Doesn't this just get us back to the same place, where `UserInput` can't be reasonably extended?

Reply:
This struct has just been moved out from the original user_input.rs file.

Reply:
It should be able to replace with a new trait for implementing input mocking

Reply:
Oh, I see!